src/EnergyPlus/EconomicLifeCycleCost.cc

```diff
@@ -1353,53 +1352,53 @@ void ExpressAsCashFlows(EnergyPlusData &state)
-                elcc->CashFlow(curCategory).mnAmount(jMonth) += elcc->CashFlow(jCost).mnAmount(jMonth);
+                elcc->CashFlow[curCategory].mnAmount(jMonth) += elcc->CashFlow[jCost].mnAmount(jMonth);
             }
         }
     }
     // create total categories
     for (int jMonth = 1; jMonth <= elcc->lengthStudyTotalMonths; ++jMonth) {
-        elcc->CashFlow(costCatTotEnergy).mnAmount(jMonth) = elcc->CashFlow(costCatEnergy).mnAmount(jMonth);
-        elcc->CashFlow(costCatTotOper).mnAmount(jMonth) =
-            elcc->CashFlow(costCatMaintenance).mnAmount(jMonth) + elcc->CashFlow(costCatRepair).mnAmount(jMonth) +
-            elcc->CashFlow(costCatOperation).mnAmount(jMonth) + elcc->CashFlow(costCatReplacement).mnAmount(jMonth) +
-            elcc->CashFlow(costCatMinorOverhaul).mnAmount(jMonth) + elcc->CashFlow(costCatMajorOverhaul).mnAmount(jMonth) +
-            elcc->CashFlow(costCatOtherOperational).mnAmount(jMonth) + elcc->CashFlow(costCatWater).mnAmount(jMonth) +
-            elcc->CashFlow(costCatEnergy).mnAmount(jMonth);
-        elcc->CashFlow(costCatTotCaptl).mnAmount(jMonth) = elcc->CashFlow(costCatConstruction).mnAmount(jMonth) +
-                                                           elcc->CashFlow(costCatSalvage).mnAmount(jMonth) +
-                                                           elcc->CashFlow(costCatOtherCapital).mnAmount(jMonth);
-        elcc->CashFlow(costCatTotGrand).mnAmount(jMonth) =
-            elcc->CashFlow(costCatTotOper).mnAmount(jMonth) + elcc->CashFlow(costCatTotCaptl).mnAmount(jMonth);
+        elcc->CashFlow[CostCategory::TotEnergy].mnAmount(jMonth) = elcc->CashFlow[CostCategory::Energy].mnAmount(jMonth);
+        elcc->CashFlow[CostCategory::TotOper].mnAmount(jMonth) =
+            elcc->CashFlow[CostCategory::Maintenance].mnAmount(jMonth) + elcc->CashFlow[CostCategory::Repair].mnAmount(jMonth) +
+            elcc->CashFlow[CostCategory::Operation].mnAmount(jMonth) + elcc->CashFlow[CostCategory::Replacement].mnAmount(jMonth) +
+            elcc->CashFlow[CostCategory::MinorOverhaul].mnAmount(jMonth) + elcc->CashFlow[CostCategory::MajorOverhaul].mnAmount(jMonth) +
+            elcc->CashFlow[CostCategory::OtherOperational].mnAmount(jMonth) + elcc->CashFlow[CostCategory::Water].mnAmount(jMonth) +
+            elcc->CashFlow[CostCategory::Energy].mnAmount(jMonth);
+        elcc->CashFlow[CostCategory::TotCaptl].mnAmount(jMonth) = elcc->CashFlow[CostCategory::Construction].mnAmount(jMonth) +
+                                                                  elcc->CashFlow[CostCategory::Salvage].mnAmount(jMonth) +
+                                                                  elcc->CashFlow[CostCategory::OtherCapital].mnAmount(jMonth);
+        elcc->CashFlow[CostCategory::TotGrand].mnAmount(jMonth) =
+            elcc->CashFlow[CostCategory::TotOper].mnAmount(jMonth) + elcc->CashFlow[CostCategory::TotCaptl].mnAmount(jMonth);
```

Comment:
Enums start at `0` so converted `elcc->CashFlow` from `EPVector` to `std::vector`.

Reply:
If `CashFlow` is really indexed by an `enum` then you can convert it to `std::array`. This applies to all arrays indexed by `enum`.

Reply:
Only part of `CashFlow` is indexed by an enum... It is more of a `std::vector`/`Array1D` like container that can [change size](https://github.com/NREL/EnergyPlus/blob/00d968c57b48ff354fdc123d00e782e18842f1ba/src/EnergyPlus/EconomicLifeCycleCost.cc#L1225) based on the number of items in the IDF. 

More details:
[As shown in this line](https://github.com/NREL/EnergyPlus/blob/00d968c57b48ff354fdc123d00e782e18842f1ba/src/EnergyPlus/EconomicLifeCycleCost.cc#L1219), numCashFlow is the size of CashFlow, : 
`elcc->numCashFlow = countOfCostCat + elcc->numRecurringCosts + elcc->numNonrecurringCost + elcc->numResourcesUsed;`. 
In that equation, `countOfCostCat` is always constant and equal to the number of enum values, and that part of `CashFlow` is indexed by the enum. The other terms (`numRecurringCosts`, `numNonrecurringCost`, `numResourcesUsed`) depend on the IDF.

Reply:
I see, thanks for that clarification. As you were.